### DISCUSSION TITLE
SQLAlchemy unnest and table_valued: Missing Column Names in AS Clause?

### DISCUSSION THREAD

**Discussion by simonnelse** [sqlalchemy/sqlalchemy#10855] [Usage Questions] [answered]
```python
from sqlalchemy import func, select
from sqlalchemy.dialects.postgresql import array

fn = func.unnest(array([1, 2, 3]), array(['A', 'B', 'C'])).table_valued("col1", "col2", name="x")
print(select(fn.c.col1, fn.c.col2))
```
Encountered an unexpected behaviour when using the `unnest` function along with the `table_valued` method. The provided code outputs the following SQL query:
```sql
SELECT x.col1, x.col2 
FROM unnest(ARRAY[%(param_1)s, %(param_2)s, %(param_3)s], ARRAY[%(param_1)s, %(param_2)s, %(param_3)s]) AS x
```

However, I was expecting the output to include column names in the `AS` clause, like this:
```sql
SELECT x.col1, x.col2 
FROM unnest(ARRAY[%(param_1)s, %(param_2)s, %(param_3)s], ARRAY[%(param_1)s, %(param_2)s, %(param_3)s]) AS x(col1, col2);
```

The resulting SQL query lacks the column names in the `AS` clause, which seems unusual to me. I'd be grateful for any insights or explanations on this behaviour – perhaps there's something I'm overlooking?

Thank you!

Python version:  3.10.12
SQLAlchemy version: 1.4.34

**Comment by CaselIT** [comment, marked as answer]
Hi,

you would have to call `render_derived` https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.TableValuedAlias.render_derived

**Reply by simonnelse** [reply]
Thank you! It did the trick.

**Reply by simonnelse** [reply]
@CaselIT On a different note, I'm curious about the reasoning behind having a separate function for this. Why do I need to specify it explicitly? If you can provide more insight or direct me to the reasons behind this design choice, I'd appreciate it!

**Reply by CaselIT** [reply]
I think the original target use case was something like `generate_series`, as mentioned in the table_valued examples.

Also I'm not sure if some db don't like the columns to be specified in the alias
